### PR TITLE
reference/configuration, faq: add both config and faq of server-version.

### PR DESCRIPTION
**Pull request by reafans**
### What is changed, added or deleted? (Required)
1. add config of `server-version`.
2. add faq of `server-version`.

### Which TiDB version(s) do your changes apply to? (Required)

- [x] master (the latest development version, including v4.0 changes for now)
- [x] v3.1 (TiDB 3.1 versions)
- [x] v3.0 (TiDB 3.0 versions)
- [x] v2.1 (TiDB 2.1 versions)

**If you select two or more versions from above**, to trigger the bot to cherry-pick this PR to your desired release version branch(es), you **must** add corresponding labels such as **needs-cherry-pick-3.1**, **needs-cherry-pick-3.0**, and **needs-cherry-pick-2.1**.

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):

### CODE REVIEW

File: faq/tidb.md [hunk 1]
```diff
@@ -113,7 +113,11 @@ TiDB 的自增 ID (`AUTO_INCREMENT`) 只保证自增且唯一，并不保证连
 
 TiDB 的 sql_mode 与 MySQL 的 sql_mode 设置方法有一些差别，TiDB 不支持配置文件配置设置数据库的 sql\_mode，而只能使用 set 命令去设置，具体方法为：`set @@global.sql_mode = 'STRICT_TRANS_TABLES';`。
 
-#### 1.1.23 TiDB 支持哪些认证协议，过程是怎样的？
+#### 1.1.23 TiDB 是否支持修改版本号？我们的安全漏洞扫描工具对 mysql version 有要求。
```
Comment (zimulala) [inline faq/tidb.md]: s/mysql/MySQL

File: reference/configuration/tidb-server/configuration-file.md [hunk 2]
```diff
@@ -83,6 +83,14 @@ TiDB 配置文件比命令行参数支持更多的选项。你可以在 [config/
 + 默认值：false
 + 默认情况下，不支持增删主键。将此变量被设置为 true 后，支持增删主键功能。不过对在此开关开启前已经存在的表，且主键是整型类型时，即使之后开启此开关也不支持对此列表删除主键。
 
+### `server-version`
+
++ 用来修改 TiDB 服务端在以下情况下返回的版本:
+    - 当使用内置函数 `VERSION()` 时。
+    - 当与客户端初始连接，TiDB 返回带有服务端版本的初始握手包时。具体可以查看 MySQL 初始握手包[描述](<https://dev.mysql.com/doc/internals/en/connection-phase-packets.html#packet-Protocol::Handshake> for more details)。
```
Comment (zimulala) [inline reference/configuration/tidb-server/configuration-file.md]: 这个超链接显示有问题

File: faq/tidb.md [hunk 3]
```diff
@@ -113,7 +113,11 @@ TiDB 的自增 ID (`AUTO_INCREMENT`) 只保证自增且唯一，并不保证连
 
 TiDB 的 sql_mode 与 MySQL 的 sql_mode 设置方法有一些差别，TiDB 不支持配置文件配置设置数据库的 sql\_mode，而只能使用 set 命令去设置，具体方法为：`set @@global.sql_mode = 'STRICT_TRANS_TABLES';`。
 
-#### 1.1.23 TiDB 支持哪些认证协议，过程是怎样的？
+#### 1.1.23 我们的安全漏洞扫描工具对 MySQL version 有要求, TiDB 是否支持修改 server 版本号呢？
+
+TiDB 在 v3.0.8 之后是支持修改 server 版本号的，可通过配置文件中的 `server-version` 配置项进行调整。在使用 tidb-ansible 部署集群时，同样可以通过 `conf/tidb.yml` 配置文件中的 `server-version` 来设置合适的版本号，以避免出现安全漏洞扫描不通过的问题。
```
Comment (bb7133) [inline faq/tidb.md]: Please add a hyperlink to https://github.com/pingcap/docs-cn/pull/2573/files#diff-03d37f540e23cadd32b8e1c4c20d4b76R86

So the users can be navigated to the details of this configuration entry smoothly.

File: reference/configuration/tidb-server/configuration-file.md [hunk 4]
```diff
@@ -83,6 +83,14 @@ TiDB 配置文件比命令行参数支持更多的选项。你可以在 [config/
 + 默认值：false
 + 默认情况下，不支持增删主键。将此变量被设置为 true 后，支持增删主键功能。不过对在此开关开启前已经存在的表，且主键是整型类型时，即使之后开启此开关也不支持对此列表删除主键。
 
+### `server-version`
+
++ 用来修改 TiDB 服务端在以下情况下返回的版本:
+    - 当使用内置函数 `VERSION()` 时。
+    - 当与客户端初始连接，TiDB 返回带有服务端版本的初始握手包时。具体可以查看 MySQL 初始握手包的[描述](https://dev.mysql.com/doc/internals/en/connection-phase-packets.html#packet-Protocol::Handshake)。
++ 默认值：""
++ 默认情况下，TiDB 服务端版本为 TiDB 的原始版本。
```
Comment (bb7133) [inline reference/configuration/tidb-server/configuration-file.md]:
```suggestion
+ 默认情况下，TiDB 版本号格式为：`5.7.${mysql_latest_minor_version}-TiDB-${tidb_version}`。
```

File: faq/tidb.md (hunk 3 above)
Comment (reafans) [reply faq/tidb.md]: Great！

File: reference/configuration/tidb-server/configuration-file.md [hunk 5]
```diff
@@ -83,6 +83,14 @@ TiDB 配置文件比命令行参数支持更多的选项。你可以在 [config/
 + 默认值：false
 + 默认情况下，不支持增删主键。将此变量被设置为 true 后，支持增删主键功能。不过对在此开关开启前已经存在的表，且主键是整型类型时，即使之后开启此开关也不支持对此列表删除主键。
 
+### `server-version`
+
++ 用来修改 TiDB 服务端在以下情况下返回的版本:
```
Comment (AilinKid) [inline reference/configuration/tidb-server/configuration-file.md]:
```suggestion
+ 用来修改 TiDB 服务端在以下情况返回的版本:
```

File: faq/tidb.md [hunk 6]
```diff
@@ -113,7 +113,11 @@ TiDB 的自增 ID (`AUTO_INCREMENT`) 只保证自增且唯一，并不保证连
 
 TiDB 的 sql_mode 与 MySQL 的 sql_mode 设置方法有一些差别，TiDB 不支持配置文件配置设置数据库的 sql\_mode，而只能使用 set 命令去设置，具体方法为：`set @@global.sql_mode = 'STRICT_TRANS_TABLES';`。
 
-#### 1.1.23 TiDB 支持哪些认证协议，过程是怎样的？
+#### 1.1.23 我们的安全漏洞扫描工具对 MySQL version 有要求, TiDB 是否支持修改 server 版本号呢？
```
Comment (TomShawn) [inline faq/tidb.md]:
```suggestion
#### 1.1.23 我们的安全漏洞扫描工具对 MySQL version 有要求，TiDB 是否支持修改 server 版本号呢？
```

File: faq/tidb.md [hunk 7]
```diff
@@ -113,7 +113,11 @@ TiDB 的自增 ID (`AUTO_INCREMENT`) 只保证自增且唯一，并不保证连
...
+
+TiDB 在 v3.0.8 之后是支持修改 server 版本号的，可以通过配置文件中的 [`server-version`](/reference/configuration/tidb-server/configuration-file.md#server-version) 配置项进行修改。在使用 tidb-ansible 部署集群时，同样可以通过 `conf/tidb.yml` 配置文件中的 `server-version` 来设置合适的版本号，以避免出现安全漏洞扫描不通过的问题。
```
Comment (TomShawn) [inline faq/tidb.md]:
```suggestion
TiDB 在 v3.0.8 后支持修改 server 版本号，可以通过配置文件中的 [`server-version`](/reference/configuration/tidb-server/configuration-file.md#server-version) 配置项进行修改。在使用 TiDB Ansible 部署集群时，同样可以通过 `conf/tidb.yml` 配置文件中的 `server-version` 来设置合适的版本号，以避免出现安全漏洞扫描不通过的问题。
```

**Comment by bb7133** [comment]
LGTM

**Comment by reafans** [comment]
@TomShawn ，done，PTAL

**Pull request merged**